### PR TITLE
Contact confirm

### PR DESCRIPTION
**Pull request by jasiek-net**
this `PR` needs to update env vars (checkout .env.example)

### CODE REVIEW

File: src/helpers/passport.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export function authenticate(req: Request, res: Response, next: NextFunction): P
   return new Promise((resolve, reject) => {
     passport.authenticate(
       'jwt',
-      { session: process.env.JWT_SESSION },
+      { session: process.env.JWT_SESSION === 'true' },
```
Comment (michalmikolajczyk) [inline src/helpers/passport.ts:35]: are .you sure this is valid?

File: src/mailer/index.ts [hunk 2]
```diff
@@ -1,55 +1,90 @@
...
+    !process.env.GMAIL_CONTACT_REFRESH_TOKEN) {
+  log('You should set Gmail OAuth2 envs');
 }
 
-const transporter = nodemailer.createTransport({
+const rideAddress = process.env.GMAIL_RIDE_USER;
+const contactAddress = process.env.GMAIL_CONTACT_USER;
+
+const transporterRide = nodemailer.createTransport({
   host: 'smtp.gmail.com',
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:27]: nope

File: src/mailer/index.ts [hunk 3]
```diff
@@ -1,55 +1,90 @@
...
   port: 465,
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:28]: should be derived from an env var

File: src/mailer/index.ts [hunk 4]
```diff
@@ -1,55 +1,90 @@
...
+    clientSecret: process.env.GMAIL_RIDE_CLIENT_SECRET,
+    refreshToken: process.env.GMAIL_RIDE_REFRESH_TOKEN,
   },
 });
 
-function sendEmail(body: any): Promise<any> {
+const transporterContact = nodemailer.createTransport({
+  host: 'smtp.gmail.com',
+  port: 465,
+  secure: true,
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:42]: just like above

File: src/mailer/index.ts [hunk 5]
```diff
@@ -59,21 +94,38 @@ function contactEmail(data: any): Promise<any> {
...
+  if (type === 'ride') {
+    const contactRequestBody = {
+      from: rideAddress,
+      to: rideAddress,
+      ...contactRequest(name, email, type, message, body),
+    };
+    const rideBody = {
+      from: rideAddress,
+      to: email,
+      ...rideMessage(name, code),
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:108]: nice style

File: src/services/contact/contact.ts [hunk 6]
```diff
@@ -30,7 +31,7 @@ export default function contact(req: Request, res: Response, next: NextFunction)
   };
 
   db['contact'].create(newContact)
-    .then(user => contactEmail(newContact)
+    .then(user => contactEmail({ ...newContact, code })
       .then(() => res.status(200).json({ msg: 'Contact request saved' }))
       .catch((err) => {
         log(`Unexpected error ${err}`);
```
Comment (michalmikolajczyk) [inline src/services/contact/contact.ts:37]: let's log error text messages, specific and descriptive to each expected error, e.g. `connection to the smtp service failed, email not set` and `contact not saved`.

consider that `caught` is just another block, and can be used between other blocks, e.g. `db['contact'].create(newContact).then().catch().then()...`

please ensure that all failed or aborted actions are reported. In a case where an operation consists of 5 steps, and the fail happened at step 3., then the logs should display the errors for step 3. and also steps 4. and 5.

**Review by michalmikolajczyk — APPROVED** [review]

File: src/helpers/passport.ts (hunk 1 above)
Comment (jasiek-net) [reply src/helpers/passport.ts:35]: yep, it's because env vars are strings

**Pull request merged**